apis/flowcollector/v1beta2/flowcollector_types.go

```diff
@@ -864,6 +864,26 @@ type AdvancedAgentConfig struct {
 	// in edge debug or support scenarios.
 	//+optional
 	Env map[string]string `json:"env,omitempty"`
+
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	// +optional
+	// +mapType=atomic
+	NodeSelector map[string]string `json:"nodeSelector,omitempty" protobuf:"bytes,7,rep,name=nodeSelector"`
```

Comment:
I would use `metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"` type for nodeSelector
`NodeSelector metav1.LabelSelector`

Reply:
As discussed I'm setting `corev1.PodSpec` attributes for each components. These take `map[string]string` for `NodeSelector`.